Missing Dependency `ament_index_cpp` Causes Build Failure in `audio_common` for Music Node

When attempting to build the audio_common package using `colcon build`, the build process fails with the following error:

```
/path/to/workspace/src/audio_common/audio_common/src/audio_common/music_node.cpp:30:10: fatal error: ament_index_cpp/get_package_share_directory.hpp: No such file or directory
   30 | #include <ament_index_cpp/get_package_share_directory.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Steps to Reproduce:

Clone the repository and navigate to the workspace.

```
git clone <repository-url>
cd <workspace>
```
Run the build command:
```
colcon build
```
The build fails with the error message mentioned above.

I was able to resolve the issue by updating the `CMakeLists.txt` file to include `ament_index_cpp`as a required dependency. The following changes fixed the issue:

Changes in `CMakeLists.txt`:
```cmake
# Added the following line to locate the package
find_package(ament_index_cpp REQUIRED)

# Updated include directories
include_directories(
  include
  ${PORTAUDIO_INCLUDE_DIR}
  ${ament_index_cpp_INCLUDE_DIRS}
)

# Updated dependencies for the target
ament_target_dependencies(music_node rclcpp std_srvs audio_common_msgs ament_index_cpp)
```

Environment:
```
ROS 2 Distro: echo $ROS_DISTRO -> rolling
Ubuntu Version: Ubuntu 24.04
```

Thank you for maintaining this project! Please let me know if you need any further details.

--- a/audio_common/CMakeLists.txt
+++ b/audio_common/CMakeLists.txt
@@ -3,6 +3,7 @@ project(audio_common)
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
+find_package(ament_index_cpp REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_action REQUIRED)
 find_package(std_srvs REQUIRED)
@@ -41,7 +42,7 @@ add_executable(music_node
   src/audio_common/music_node.cpp
   src/music_main.cpp
 )
-ament_target_dependencies(music_node rclcpp std_srvs audio_common_msgs)
+ament_target_dependencies(music_node rclcpp std_srvs audio_common_msgs ament_index_cpp)
 target_link_libraries(music_node ${PORTAUDIO_LIB})
 install(TARGETS
   music_node
@@ -56,7 +57,7 @@ add_executable(tts_node
 ament_target_dependencies(tts_node rclcpp rclcpp_action audio_common_msgs)
 target_link_libraries(tts_node ${PORTAUDIO_LIB})
 install(TARGETS
-tts_node
+  tts_node
   DESTINATION lib/${PROJECT_NAME}
 )
 

--- a/audio_common/package.xml
+++ b/audio_common/package.xml
@@ -12,6 +12,7 @@
   <test_depend>ament_lint_auto</test_depend>
   <test_depend>ament_lint_common</test_depend>
 
+  <depend>ament_index_cpp</depend>
   <depend>audio_common_msgs</depend>
   <depend>portaudio19-dev</depend>
   <depend>espeak</depend>

